assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -35,8 +38,55 @@ module algorithm_info_mod
 integer, parameter :: BOUNDED_NORMAL_RHF = 101 
 
 public :: obs_error_info, probit_dist_info, obs_inc_info, &
+          init_algorithm_info_mod, end_algorithm_info_mod, &
           EAKF, ENKF, BOUNDED_NORMAL_RHF, UNBOUNDED_RHF, GAMMA_FILTER
 
+!Creates the type definitions for the QCF table
+type obs_error_info_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_inflation_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_state_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_extended_state_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type obs_inc_info_type
+   character(len=129)    :: filter_kind
+   logical               :: rectangular_quadrature, gaussian_likelihood_tails
+   logical               :: sort_obs_inc, spread_restoration
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type qcf_table_data_type
```

Comment:
Similar naming comment:

I think this type should get a name that describes what it is, rather than qcf_table.. 

qcf_table_data_type - internal to algorithm_info_mod, this is not a table

Maybe something like:
algorithm_info_type